pkg/domain/plan_replayer_slow_log_test.go

```diff
@@ -0,0 +1,106 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain_test
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/config"
+	"github.com/pingcap/tidb/pkg/testkit"
+	"github.com/pingcap/tidb/pkg/util/logutil"
+	"github.com/stretchr/testify/require"
+)
+
+// TestPlanReplayerInternalQuery verifies that queries from plan replayer are marked as internal.
+// This test checks both slow log (Is_internal field) and statement summary filtering.
+func TestPlanReplayerInternalQuery(t *testing.T) {
+	// Setup slow log file
+	f, err := os.CreateTemp("", "tidb-slow-*.log")
+	require.NoError(t, err)
+	slowLogFile := f.Name()
+	require.NoError(t, f.Close())
+	defer os.Remove(slowLogFile)
+
+	// Configure slow log
+	originCfg := config.GetGlobalConfig()
+	newCfg := *originCfg
+	newCfg.Log.SlowQueryFile = slowLogFile
+	config.StoreGlobalConfig(&newCfg)
+	defer config.StoreGlobalConfig(originCfg)
+
+	require.NoError(t, logutil.InitLogger(newCfg.Log.ToLogConfig()))
+
+	store, dom := testkit.CreateMockStoreAndDomain(t)
+	tk := testkit.NewTestKit(t, store)
+
+	// Enable slow log with threshold = 0 to capture all queries
+	tk.MustExec("set global tidb_slow_log_threshold = 0")
+	tk.MustExec("set global tidb_enable_slow_log = 1")
+	tk.MustExec(fmt.Sprintf("set @@tidb_slow_query_file='%v'", slowLogFile))
+
+	// Configure statement summary to NOT record internal queries
+	tk.MustExec("set global tidb_enable_stmt_summary = 1")
+	tk.MustExec("set global tidb_stmt_summary_refresh_interval = 3600")
+	tk.MustExec("set global tidb_stmt_summary_internal_query = 0")
+
+	// Setup test data
+	tk.MustExec("delete from mysql.plan_replayer_task")
+	tk.MustExec("insert into mysql.plan_replayer_task (sql_digest, plan_digest) values ('test_digest', 'test_plan')")
+
+	// Trigger plan replayer collection
+	prHandle := dom.GetPlanReplayerHandle()
+	err = prHandle.CollectPlanReplayerTask()
+	require.NoError(t, err)
+
+	// Wait for slow log to be written
+	time.Sleep(500 * time.Millisecond)
+
+	// Check slow log for Is_internal field
+	slowLogContent, err := os.ReadFile(slowLogFile)
+	if err == nil && len(slowLogContent) > 0 {
+		slowLogStr := string(slowLogContent)
+		lines := strings.Split(slowLogStr, "\n")
+
+		for i, line := range lines {
+			if strings.Contains(line, "plan_replayer_task") && strings.Contains(strings.ToLower(line), "select") {
+				// Found the SELECT query, look for Is_internal field
+				for j := i + 1; j < len(lines) && j < i+30; j++ {
```

Comment:
It seems that this is checking the `Is_internal` field of the statement following the query `SELECT sql_digest, plan_digest FROM mysql.plan_replayer_task;`. 
I'm not sure if this process fully matches the intended test scenario, but the result does indeed meet expectations.

Rest LGTM